nixos/tests/slurm.nix

```diff
@@ -126,8 +126,17 @@ in {
 
   testScript =
   ''
+  # There's currently no way for slurmdbd to signal its readyness to slurmctld.
+  # If slurmctld happens to start before slurmdbd, it errors out.
```

Comment:
Omg, this is so frustrating, if I put slurmdbd and slurmctld on the same node, it fails again: slurmctld begins to start as part of `multi-user.target`, and even if it has `after = [ "slurmdbd.service" ]`, by the time `slurmdbd` is actually ready, the slurmctld has already tried connecting to it and failed

Reply:
I actually have both running on the same machine and this has never been a problem so far.

Reply:
Fails in the nixosTests.slurm for me. Could you check if slurmctld may have failed on you and got restarted by systemd? What crashes the nixosTests is the wait_for_unit, which doesn't care if the service is restarted later

Reply:
The test runs fine on current nixpkgs/master. Under which circumstances does it fail?

Reply:
> The test runs fine on current nixpkgs/master. 

You do need to move `srun` or `wait_for_unit` up. On master the test first issues the `systemctl restart`

Reply:
`nix build -L github:NixOS/nixpkgs/df97ef0253a41d03c4ad71055d3c0524a5c3d013#nixosTests.slurm`

I get `slurmctld[774]: fatal: You are running with a database but for some reason we have no TRES from it.  This should only happen if the database is down and you don't have 
any state files` somewhere in the middle of the logs, and then later `srun` failing

Cf. https://github.com/NixOS/nixpkgs/pull/267936/commits/df97ef0253a41d03c4ad71055d3c0524a5c3d013

Reply:
You might have to `--rebuild` a few times